pb/edge_ctrl_pb/edge_ctrl.proto

```diff
@@ -3,182 +3,240 @@ syntax = "proto3";
 
-    message Process {
-        repeated string paths = 1;
-        HashAlgorithm hashAlgorithm = 10;
-    }
+  message Process {
+    repeated string paths = 1;
+    HashAlgorithm hashAlgorithm = 10;
+  }
 }
 
 
 message PostureResponse {
-    string id = 1;
-    oneof payload {
-        PostureResponse.Error error = 20;
-        PostureResponse.Proc process = 30;
-        PostureResponse.MacAddresses mac = 40;
-        PostureResponse.WindowsDomain domain = 50;
-        PostureResponse.OperatingSystem os = 60;
+  string id = 1;
+  oneof payload {
+    PostureResponse.Error error = 20;
+    PostureResponse.Proc process = 30;
+    PostureResponse.MacAddresses mac = 40;
+    PostureResponse.WindowsDomain domain = 50;
+    PostureResponse.OperatingSystem os = 60;
+  }
+
+
+  message Error {
+    enum Code {
+      UNKNOWN = 0;
+      UNSUPPORTED_QUERY_TYPE = 10;
+      UNSUPPORTED_HASH_ALGORITHM = 20;
     }
 
+    PostureResponse.Error.Code code = 1;
+    string message = 10;
+    map<string, string> data = 20;
+  }
 
-    message Error {
-        enum Code {
-            UNKNOWN = 0;
-            UNSUPPORTED_QUERY_TYPE = 10;
-            UNSUPPORTED_HASH_ALGORITHM = 20;
-        }
+  message Proc {
+    string binaryHash = 1;
+    string signerFingerprint = 10;
+  }
 
-        PostureResponse.Error.Code code = 1;
-        string message = 10;
-        map<string, string> data = 20;
-    }
+  message MacAddresses {
+    repeated string addresses = 1;
+  }
 
-    message Proc {
-        string binaryHash = 1;
-        string signerFingerprint = 10;
-    }
+  message WindowsDomain {
+    string name = 1;
+  }
 
-    message MacAddresses {
-        repeated string addresses = 1;
-    }
+  message OperatingSystem {
+    PostureResponse.OperatingSystem.Type type = 1;
+    string version = 10;
+    string build = 20;
 
-    message WindowsDomain {
-        string name = 1;
+    enum Type {
+      Android = 0;
+      iOS = 10;
+      Linux = 20;
+      macOS = 30;
+      Windows = 40;
+      WindowsServer = 50;
     }
+  }
+}
 
-    message OperatingSystem {
-        PostureResponse.OperatingSystem.Type type = 1;
-        string version = 10;
-        string build = 20;
-
-        enum Type {
-            Android = 0;
-            iOS = 10;
-            Linux = 20;
-            macOS = 30;
-            Windows = 40;
-            WindowsServer = 50;
-        }
-    }
+message CreateCircuitRequest {
+  string sessionToken = 1;
+  repeated string fingerprints = 2;
+  string terminatorIdentity = 3;
+  map<uint32, bytes> peerData = 4;
+}
+
+message CreateCircuitResponse {
+  string sessionId = 1;
+  string address = 2;
+  map<uint32, bytes> peerData = 3;
+}
+
+enum TerminatorPrecedence {
+  Default = 0;
+  Required = 1;
+  Failed = 2;
+}
+
+message CreateTerminatorRequest {
+  string sessionToken = 1;
+  repeated string fingerprints = 2;
+  map<uint32, bytes> peerData = 3;
+  uint32 cost = 4;
+  TerminatorPrecedence precedence = 5;
+  string identity = 6;
```

Comment:
is this `identityId` ?

Reply:
No it's a fabric terminator identifier used for grouping terminators. Maybe renamed to group?